cpp/python/pystrings.cpp

```diff
@@ -135,6 +136,56 @@ bool parse_args( const char* fn, PyObject* pyargs, const char* pyfmt, ... )
     return rtn;
 }
 
+static std::basic_string<char> ConvertCudaIpcMemHandler ( cudaIpcMemHandle_t ipc_memhandle )
+{
+    std::basic_string<char> bytes;
+    bytes.resize(sizeof(cudaIpcMemHandle_t));
+    memcpy((void*)bytes.data(), (char*)(&ipc_memhandle), sizeof(cudaIpcMemHandle_t));
+    return bytes;
+}
+
+static cudaIpcMemHandle_t ConvertByteArray ( PyObject* bytearray )
+{
+    char* string_array = PyByteArray_AsString(bytearray);
+    cudaIpcMemHandle_t ipc_memhandle;
+    memcpy((char*)&ipc_memhandle, string_array, sizeof(cudaIpcMemHandle_t));
+    return ipc_memhandle;
+}
+
+static PyObject* n_createFromIPC( PyObject* self, PyObject* args )
```

Comment:
I think that a better implementation here is just to copy the entire ipc into a single byte-array rather than break it up into its components. There is no reason to access the individual parts in python. The only way to do anything with the items is to call C methods.
lines 179-186 can be replaced with:
```
return PyByteArray_FromStringAndSize(&ipc,sizeof(ipc));
```
then replace lines 159-165 with
```
memcpy(&ipc, PyByteArray_AsString(PyTuple_GetItem(args,0), sizeof(ipc));
```
You can also remove the 2 `Convert` methods then as well.
This approach allows changes to the ipc_transfer structure without further changes to this code.
On the server/send side you can use the above to create ipc-transfer and get access to the handles.

Reply:
The redundant code was replaced. Good catch, thank you.